osversion/platform_compat_windows_test.go

```diff
@@ -0,0 +1,60 @@
+package osversion
+
+import (
+	"testing"
+)
+
+const (
+	RS5Prefix         = "10.0.17763"
+	V21H2ServerPrefix = "10.0.20348"
+	V22H2Win11Prefix  = "10.0.22621"
+)
+
+// Test the platform compatibility of the different
+// OS Versions considering two ltsc container image
+// versions (ltsc2019, ltsc2022)
+func Test_PlatformCompat_HostWS2022(t *testing.T) {
+	for _, tc := range map[string]struct {
+		host      string
+		guestOs   string
+		shouldRun bool
+	}{
+		"WS2022Host_ltsc2022": {
+			host:      V21H2ServerPrefix,
+			guestOs:   V21H2ServerPrefix,
+			shouldRun: true,
+		},
+		"WS2022Host_ltsc2019": {
+			host:      V21H2ServerPrefix,
+			guestOs:   RS5Prefix,
+			shouldRun: false,
+		},
+		"RS5Host_ltsc2022": {
+			host:      RS5Prefix,
+			guestOs:   V21H2ServerPrefix,
+			shouldRun: false,
+		},
+		"RS5Host_ltsc2019": {
+			host:      RS5Prefix,
+			guestOs:   RS5Prefix,
+			shouldRun: true,
+		},
+		"Wind11Host_ltsc2022": {
+			host:      V22H2Win11Prefix,
+			guestOs:   V21H2ServerPrefix,
+			shouldRun: true,
+		},
+		"Wind11Host_ltsc2019": {
+			host:      V22H2Win11Prefix,
+			guestOs:   RS5Prefix,
+			shouldRun: false,
+		},
+	} {
+		// Check if ltsc2019/ltsc2022 guest images are compatible on
+		// the given host OS versions
+		//
+		if CanRun(tc.host, tc.guestOs) != tc.shouldRun {
+			t.Fatalf("cannot run %v guest image on host %v", tc.guestOs, tc.host)
```

Comment:
This error message can be improved, since it doesn't clearly reflect what the value of `tc.shouldRun` was for the test case.

Maybe:

```golang
var s string
if !tc.shouldRun {
    s = " NOT"
}
t.Fatalf("host %v should%s be able to run guest %v", tc.host, s, tc.guestOs)
```